active-rfcs/0000-finalize-head-body.md

```diff
@@ -0,0 +1,93 @@
+- Reference Issues: https://github.com/withastro/rfcs/discussions/15
+- Implementation PR:
+
+# Summary
+
+Finalize behavior around `<head>`, `<body>`, `<html>`, and `<!DOCTYPE html>` elements in Astro component templates.
+
+# Motivation
+
+in Astro v0.21, we have had user reports around `<head>` not acting as expected, breaking some projects from v0.20:
+- https://github.com/withastro/astro/issues/2046
+- https://github.com/withastro/astro/issues/2022
+
+Some of these issues stem from undocumented or undefined behaviors in v0.20, like the fact that `<head>` could be conditional.
+
+This RFC is an attempt to agree on and document the behavior of the `<head>`, `<body>`, and `<html>` tags in a way that makes sense, and can be communicated to users. It should support most common use-cases, such as layouts:
+
+```astro
+---
+// src/pages/index.astro
+import CommonLayout from '../layouts/CommonLayout.astro';
+---
+<!-- this is valid, where CommonLayout contains `<head>`, `<body>`, etc.
+<CommonLayout someProp="..." />
+```
+
+## Goals
+
+- Agree on and document the rules and behaviors of the `<head>`, `<body>`, and `<html>` tags.
+- Support any component in the tree adding to the page `<head>`, similar to `<svelte:head>` or head manager libraries in React/Vue. 
+  - (Note: see "Alternatives" for an alternative design if this is decided to be a non-goal).
+
+## Non-Goals
+
+- **Intentionally not implemented:** The ability or option to omit `<body>`, `<head>`, or `<html>` in the final output. This is intentionally not supported here in an effort to reduce complexity around when these tags might or might not be output in your final page HTML. As an Astro user, we assume that you are are comfortable with Astro controlling/optimizing your final HTML output.
+  - If this is sticking point, we can reword this from "intentionally not supported" to "out of scope / possible to implement in the future", but I think it's useful to commit to this behavior as a team if there is no objection.
+- **Out of scope / future:** The ability to author without `<body>`, `<head>`, or `<html>`, and relying on the compiler to handle them like the browser would.  This makes some sense because this is valid in HTML and we try to match HTML semantics whenever possible.  This is currently treated as out-of-scope, but there is nothing here that blocks our ability to do this in the future.
+  - For example, our compiler could automatically detect some elements in the component template (for example, a `<title>`) and auto-hoist it out of the component body and into the `<head>` of the final compiler result to support this.
+- **Out of scope / future:** Intelligent merging of repeated head elements. This is common when a top-level layout `<head>` might define a title, and then a more specific component might overwrite with a more specific `<title>` of its own. This is considered out of scope but not blocked to be implemented in the future.
+
+# Detailed design
+
+- `<head>`
```

Comment:
It would be nice if these points where numbered, so you could reference them elsewhere, acting as ids ("clause 1.5 says....")

Reply:
Suprisingly, GitHub's nested numbered list support is... non-existent? Had to hack it a bit but added numbers where I could.